test/upgrade/probe_test.go

```diff
@@ -57,17 +60,61 @@ func TestContinuousEventsPropagationWithProber(t *testing.T) {
 
 	// Use zap.SugarLogger instead of t.Logf because we want to see failures
 	// inline with other logs instead of buffered until the end.
-	log := createLogger()
-	probe := prober.RunEventProber(context.Background(), log, client, config)
+	Log = createLogger()
+	p := prober.NewProber(Log, client, config)
+	p.Deploy(context.Background())
```

Comment:
Since you're making changes to signatures, I think we should plumb context through. Here we create a context.Background, and then in L76 we create another one. I think it would be better to create a context for the test and wire it through, might make things like shutdowns, cancels / timeouts easier, etc.